addons/website/static/src/snippets/s_image_comparison/options.js

```diff
@@ -0,0 +1,159 @@
+import {_t} from 'web.core';
+
+options.registry.ImageComparison = options.Class.extend({
+    /**
+     * @override
+     */
+    start() {
+        this._super(...arguments);
+        this.__onDivClick = this._onDivClick.bind(this);
+        // This is needed because when the user clicks on the image comparison,
+        // the input capture the event, so we have to manually activate the
+        // image options.
+        this.$target[0].querySelector('div').addEventListener('click', this.__onDivClick);
+    },
+    /**
+     * @override
+     */
+    cleanForSave() {
+        const slideValue = 50;
+        this.$target[0].querySelector('input[type="range"]').value = slideValue;
+        this.$target[0].querySelector('.media_left').style.clipPath =
+            `polygon(0 0,${slideValue}% 0,${slideValue}% 100%, 0 100%)`;
+        delete this.$target[0].closest('.s_image_comparison').dataset.editedMedia;
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        this.$target[0].querySelector('div').removeEventListener('click', this.__onDivClick);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Creates or removes the captions.
+     *
+     * @see this.selectClass for parameters
+     */
+    showCaptions(previewMode, widgetValue, params) {
+        const descriptionEls = this.$target[0].querySelectorAll('.description');
+        if (widgetValue && !descriptionEls.length) {
+            // Create the two caption elements.
+            const inputEl = this.$target[0].querySelector('input[type="range"]');
+            for (const side of ['left', 'right']) {
+                const divEl = document.createElement('div');
+                divEl.className = `description ${side}-description`;
+                const spanEl = document.createElement('span');
+                spanEl.className = 'o_default_snippet_text';
+                spanEl.innerText = side === 'left' ? 'BEFORE' : 'AFTER';
+                divEl.appendChild(spanEl);
+                inputEl.after(divEl);
+            }
+        } else if (!widgetValue && descriptionEls.length) {
+            for (const el of descriptionEls) {
+                el.remove();
+            }
+        }
+    },
+    /**
+     * Creates or removes the enlarge button.
+     *
+     * @see this.selectClass for parameters
+     */
+    showZoomButton(previewMode, widgetValue, params) {
+        const buttonEl = this.$target[0].querySelector('.enlarge-button');
+        if (widgetValue && !buttonEl) {
+            // Create the button.
+            const newButtonEl = document.createElement('button');
+            newButtonEl.className = 'btn enlarge-button mr-1 mb-1';
+            const iconEl = document.createElement('i');
+            iconEl.className = 'fa fa-expand';
```

Comment:
Maybe should be `o_not_editable` unless we should be able to replace the icon.
What about giving it a tooltip ?